lab_5_scrapper/scrapper.py

```diff
@@ -90,75 +214,151 @@ def __init__(self, config: Config) -> None:
+                    self.urls.append(url)
 
     def find_articles(self) -> None:
         """
         Finds articles
         """
-        pass
+        chrome_options = ChromeOptions()
+        chrome_options.add_argument('--start-maximized')
+        if self.config.get_headless_mode():
+            chrome_options.add_argument('--headless=new')
+
+        driver = webdriver.Chrome(options=chrome_options)
+        driver.get(self.config.get_seed_urls()[0])
+
+        scroll_pause_time = 2
+        last_height = driver.execute_script("return document.body.scrollHeight")
+        while len(self.urls) < self.config.get_num_articles():
+            driver.execute_script("window.scrollTo(0,document.body.scrollHeight)")
+            current_html = driver.page_source
+            bs = BeautifulSoup(current_html, 'lxml')
+            self._extract_url(bs)
+
+            time.sleep(scroll_pause_time)
+            new_height = driver.execute_script("return document.body.scrollHeight")
+            if new_height == last_height:
+                break
+            last_height = new_height
+        driver.quit()
 
     def get_search_urls(self) -> list:
         """
         Returns seed_urls param
         """
-        pass
+        return self.config.get_seed_urls()
 
 
 class HTMLParser:
     """
     ArticleParser implementation
     """
 
-    def __init__(self, full_url: str, article_id: int, config: Config) -> None:
+    def __init__(self, article_url: str, article_id: int, config: Config) -> None:
         """
         Initializes an instance of the HTMLParser class
         """
-        pass
+        self.article_url = article_url
+        self.article_id = article_id
+        self.config = config
+        self.article = Article(url=self.article_url, article_id=self.article_id)
 
     def _fill_article_with_text(self, article_soup: BeautifulSoup) -> None:
         """
         Finds text of article
         """
-        pass
+        # get the title
+        title = article_soup.find_all('h1', {'class': 'new-title'})[0]
+        title_text = title.text
+
+        # get the preview
+        preview = article_soup.find_all('div', {'class': 'preview-text'})[0]
+        preview_text = preview.text.strip()
+        if preview_text[-1] not in '.?!':
+            preview_text += '.'
+
+        # get the article body
+        body_bs = article_soup.find_all('div', {'class': 'detail'})[0]
+        paragraph_texts = [par.text.strip() for par in body_bs.find_all('p')]
+        article_text = ' '.join(paragraph_texts)
+
+        self.article.title = title_text
+        self.article.text = preview_text + ' ' + article_text
 
     def _fill_article_with_meta_information(self, article_soup: BeautifulSoup) -> None:
         """
         Finds meta information of article
         """
-        pass
+        try:
+            author = article_soup.find_all('span', {'class': 'author'})
+            self.article.author.extend(author[0].text[7:].split(', '))
```

Comment:
what is this magic 7 about?

Reply:
right, now i've noticed that i can access the author's name without any other words